server/interfaces/interfaces.go

```diff
@@ -226,6 +226,9 @@ type Cache interface {
 	// Low level interface used for seeking and stream-writing.
 	Reader(ctx context.Context, r *resource.ResourceName, offset, limit int64) (io.ReadCloser, error)
 	Writer(ctx context.Context, r *resource.ResourceName) (CommittedWriteCloser, error)
+
+	// SupportsCompressor returns whether the cache supports storing data compressed with the given compressor
+	SupportsCompressor(compressor repb.Compressor_Value) bool
```

Comment:
is this PR backwards compatible so that if one server instance returns `true` during a rollout but another one returns `false`, things will function OK? thinking mainly about ddisk where one server instance may proxy a read/write to another server instance that has a potentially different return value for this.

Reply:
No it's not. Changed distributed_cache::SupportsCompressor to return false and I'll implement for distributed cache in a follow-up PR. Makes this PR a little too big to add it in here

Reply:
Added support for distributed cache with these 2 PRs:
https://github.com/buildbuddy-io/buildbuddy/pull/2956
https://github.com/buildbuddy-io/buildbuddy/pull/2962